portal/app/algorithms/battery_load_control.py

```diff
@@ -0,0 +1,52 @@
+import json, datetime, numpy as np
+from app.farm_updater import sonnen_api
+from django.conf import settings
+
+
+def batt_dispatch():
+    from app.models import FarmDevice, FarmData, DeviceType
+
+    try:
+        egauge_device = FarmDevice.objects.get(device_uid=settings.EGAUGE_ID)
+    except Exception as exc:
+        print('Error: ', exc)
+
+    farmdata = FarmData.objects.filter(farm_device=egauge_device).order_by('-id')[0:3]
```

Comment:
You can't just catch the exception, not do anything, and then still use the `egauge_device` in this query - that doesn't make sense. Exit the function is the only sensible thing to do.

Reply:
yep, I knew i was missing something here